sway-core/src/semantic_analysis/module.rs

```diff
@@ -604,7 +604,7 @@ impl ty::TyModule {
             }
 
             // Auto impl debug traits only if they are not explicitly implemented
-            if auto_impl_debug_traits {
+            if ctx.experimental.dbg_intrinsic && auto_impl_debug_traits {
```

Comment:
I have a very mixed feelings of having a feature flag that is only partially implemented here.

If we were originally for having `dbg_intrinsic` behind a feature flag that would mean:
- emitting `FeatureIsDisabled` error on `__dbg` usages.
- having the `Debug` trait and related structs (`Format`, `DebugStruct`, ...) and impls behind the flag etc.

Now we still have the penalty of always implementing script-generated `Debug` impls and adding them to the trait map even if the feature is disabled. I guess only this will still contribute to the compilation time due to huge number of types that impl `Debug`.

I understand that we just want a quick fix for the LSP, but as I said, very mixed feelings.

Reply:
Resolved, because the feature flag is reverted.